src/Core/ServiceCollectionExtensions.cs

```diff
@@ -19,7 +19,7 @@ public static class ServiceCollectionExtensions
         public static IGraphQLBuilder AddGraphQL(this IServiceCollection services, GraphQLOptions options)
         {
             services.TryAddSingleton<IDocumentExecuter, DocumentExecuter>();
-            services.AddTransient(typeof(IGraphQLExecuter<>), typeof(DefaultGraphQLExecuter<>));
+            services.AddTransient(typeof(IGraphQLExecuter<>), options.GraphQLExecuterType);
```

Comment:
I am thinking this should be instead changed to `TryAddTransient` vs. adding the type to the options.  That way you can control it with your IoC container vs. adding an option (just like the `IDocumentExecuter`).

Reply:
What would that look like? Since DefaultGraphQLExecutor is generic, I tried this, which compiles but it breaks at runtime. `services.TryAddTransient(typeof(IGraphQLExecuter), typeof(DefaultGraphQLExecuter<>));`

Reply:
Ignore that. Literally just need to add the word `Try` to the existing code haha.